src/wp-includes/class-wp-query.php

```diff
@@ -3099,8 +3099,8 @@ public function get_posts() {
 				$last_changed .= wp_cache_get_last_changed( 'terms' );
 			}
 
-			$cache_key = "wp_query:$key:$last_changed";
-
+			$cache_key      = "wp_query:$key:$last_changed";
+			$cached_results = false;
```

Comment:
```suggestion
			$cache_found = false;
```

Rather than presuming based on the return value, using the `$found` parameter would be better. This allows for falsey values to be returned without causing incorrect results. I think that can currently happen if the cached results are an empty array.

If it isn't possible now, it's a good defensive coding technique none-the-less.

:1234: The variable name will need to change throughout.

Reply:
Nitpick: I would have set this to '' (an empty string) for the default value, so the type of the variable doesn't change.

```suggestion
			$cache_found = '';
```

Reply:
You know, I have never used the found param on this function. I like it. I have refactored the PR to use it.